packages/core/app/mirage/routes/report.js

```diff
@@ -8,17 +8,39 @@ export default function() {
    * reports/ - GET endpoint to fetch many reports
    */
   this.get('/reports', function({ reports }, request) {
+    let reportsObject = null;
     let idFilter = request.queryParams['filter[reports.id]'];
+    let queryFilter = request.queryParams['filter[reports]'];
 
     // Allow filtering
     if (idFilter) {
       let ids = idFilter.split(',');
-      reports = reports.find(ids);
+      reportsObject = reports.find(ids);
+    } else if (queryFilter) {
+      let filterParameters = queryFilter
+        .split(';')[0]
```

Comment:
it's hard for me to tell first off what this transformation is doing. Would it be helpful to have a separate function that does the transformation. Then we could have a javadoc style comment at the top that says e.g.`Transforms filters in the format of xyz into abc`